x-pack/plugins/spaces/public/management/spaces_grid/spaces_grid_page.tsx

```diff
@@ -350,39 +385,80 @@ export class SpacesGridPage extends Component<Props, State> {
       }),
       actions: [
         {
-          render: (record: Space) => (
-            <EuiButtonIcon
-              data-test-subj={`${record.name}-editSpace`}
-              aria-label={i18n.translate(
-                'xpack.spaces.management.spacesGridPage.editSpaceActionName',
-                {
-                  defaultMessage: `Edit {spaceName}.`,
-                  values: { spaceName: record.name },
-                }
-              )}
-              color={'primary'}
-              iconType={'pencil'}
-              {...reactRouterNavigate(this.props.history, this.getEditSpacePath(record))}
-            />
-          ),
+          isPrimary: true,
+          name: i18n.translate('xpack.spaces.management.spacesGridPage.editSpaceActionName', {
+            defaultMessage: `Edit`,
+          }),
+          description: (rowRecord) =>
+            i18n.translate('xpack.spaces.management.spacesGridPage.editSpaceActionDescription', {
+              defaultMessage: `Edit {spaceName}.`,
+              values: { spaceName: rowRecord.name },
+            }),
+          type: 'icon',
+          icon: 'pencil',
+          color: 'primary',
+          href: (rowRecord) =>
+            reactRouterNavigate(this.props.history, this.getEditSpacePath(rowRecord)).href,
+          onClick: (rowRecord) =>
+            reactRouterNavigate(this.props.history, this.getEditSpacePath(rowRecord)).onClick,
+          'data-test-subj': (rowRecord) => `${rowRecord.name}-editSpace`,
         },
         {
-          available: (record: Space) => !isReservedSpace(record),
-          render: (record: Space) => (
-            <EuiButtonIcon
-              data-test-subj={`${record.name}-deleteSpace`}
-              aria-label={i18n.translate(
-                'xpack.spaces.management.spacesGridPage.deleteActionName',
-                {
-                  defaultMessage: `Delete {spaceName}.`,
-                  values: { spaceName: record.name },
-                }
-              )}
-              color={'danger'}
-              iconType={'trash'}
-              onClick={() => this.onDeleteSpaceClick(record)}
-            />
-          ),
+          isPrimary: true,
+          name: i18n.translate('xpack.spaces.management.spacesGridPage.switchSpaceActionName', {
+            defaultMessage: 'Switch',
```

Comment:
@kevinsweet 

You brought up a point about an overall goal to enforce more intentional navigation paths. In that regard, we maybe don't want to include a "Switch" action here to switch the user to another Space. We already have trained users to use the Space Selector: the drop-down in the top nav.

I would also mention one difference between how the Space Selector works:
* When the user changes their Space using the Space Selector, they are navigated to the "Space landing page" for that space
* When the user changes their Space using this action, the navigation would keep them in the Spaces Management page.

I'll put this PR in Draft mode for us to discuss. I'd like to avoid having to "undo" things as we roll out the project. :)

Reply:
I personally would vote for keeping the switch here too, because as Tim mentioned, it doesn't take the user away from their context or otherwise detract, given the way it's implemented here. 

But also see your point Kevin. NOT having it here should not be a big problem either.

- One more thing to consider: if we remove this option, we're left with only 2 more: edit and delete. Hence, we can remove ellipses menu as well, and save some clicks (we were there before , I know, but just not to forget about it)

Reply:
I'll reach out to the folks involved in this project, and we'll use this comment to vote and make a decision.

👍🏻 = keep the "Switch" action
👎🏻 = remove it

Reply:
Upvoted to keep the switcher.  IMO, while true this is a duplication of function, the user looking at the list of spaces is operating ON that list.  Whether aware or not of the "active space" concept, their focus is on the list, not the breadcrumb bar.  I'll draw a comparison to the Maximize button vs. double click on a title bar.  Both accomplish the same thing.  While dragging a window, the focus is on the title bar.  Sure the user could drop it, navigate to the maximize button, and use that, but it's a cumbersome, disjointed experience.  

Also, I believe this was discussed with Ryan during the design phase prior to engineering getting started and a decision was reached.  That's why it appears in the design as it does.   🙂. Unless there is a dang good reason to change the design and jeopardize the MVP implementation, let's stick to the design.

Reply:
Hmm... I don't follow. What's the point of a user being able to switch spaces and still be in space settings? It's the same screen, same menu with all the same options and controls, but now i'm in a different space in the background?

It's odd to mix these usually separate things "administration" and "navigation."

Reply:
I agree it is odd to mix them, but I think our case is a bit unusual.  In our case the space is sort of a non hierarchical third dimension or "layer", and since the active space is different from all the others, it just... "feels" right to be able to control it from the list of spaces.  Not having it there feels like something is missing and the normal space switcher combo feels too "far" from my focus.  These are just my gut opinions, but it looks like the majority of folks have the same gut.

Reply:
Thanks all, for the discussion. I believe it was worth it: got to voice concerns and listen. We have a result from the vote: the Switch action will stay.

Reply:
Nice to see this one moving fwd at pace. Bigger picture wise I'm totally on board, but in future situations, we definitely need to be able to answer the question, "what exactly **is** distinct purpose of a given UI element, from the user point of view?" Otherwise we're just adding ui clutter/complexity for no reason. But ack: that kind of consideration should happen earlier in the process prior to dev, for sure 👍 Onward and upward - excited for this one 🚀